src/Table/index.js

```diff
@@ -84,68 +84,57 @@ StyledTable.defaultProps = {
   theme: defaultTheme,
 };
 
-class Table extends React.PureComponent<Props, State> {
-  state = {
-    showShadows: false,
-    showRight: false,
-    showLeft: false,
+const Table = ({ children, compact = false, dataTest }: Props) => {
+  const [shadows, setShadows] = React.useState(false);
+  const [right, setRight] = React.useState(false);
+  const [left, setLeft] = React.useState(false);
+
+  const outer: { current: any | HTMLElement } = React.useRef(null);
+  const inner: { current: any | HTMLElement } = React.useRef(null);
+  const table: { current: any | HTMLElement } = React.useRef(null);
+
+  const handleScroll = () => {
+    if (shadows && inner && table && outer) {
+      setLeft(inner.current?.scrollLeft >= 5);
+      setRight(
+        inner.current?.scrollLeft + outer.current?.clientWidth <= table.current?.clientWidth,
```

Comment:
Also I've noticed that when you fully scroll to the right, the right shadow won't disappear. Can you check? 🙏